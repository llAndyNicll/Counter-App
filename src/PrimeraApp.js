import React from "react";
import PropTypes from 'prop-types';

// import { Fragment } from "react";



// Functional Component F.C.

const PrimeraApp = ({ saludo, subtitulo }) => {

    // Forma 1. Para validar el props.

//  if( !saludo ) {
//      throw new Error(' El saludo es necesario');
//  };


    // Puede Mostrar En Etiquetas

//  const numeros = 123;
//  const textos = 'Hola';
//  const arreglos = [ 'Tanjiro', 123 ];


    // Exepto Este Que Cuesta Un Poco 

//  const objetos = {

//     nombre:   'Adrian',
//     apellido: 'Ortega',
//     edad:     18
//  };



    return (
        <div>

            <h1> { saludo } </h1>
            <p>{ subtitulo }</p>

        </div>



        // Lo Que Se Pude Mostrar

//      <div>

//          <h1> { numeros } </h1>
//          <h1> { textos } </h1>
//          <h1> { arreglos } </h1>
//          <pre> { JSON.stringify( objetos, null, 1 ) } </pre>

//      </div>



        // Para Eliminar el DIV

//      <Fragment>
//
//          <h1> Hola Mundo </h1>
//          <p>mi primera aplicacion</p>
//
//      </Fragment>  

        // Otra Opcion Para Eliminar el DIV

//      <>
//
//          <h1> Hola Mundo </h1>
//          <p>mi primera aplicacion</p>
//
//      </>  

    );

};

// Forma 2. Para validar el props.

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    subtitulo: PropTypes.string.isRequired
};

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
};

export default PrimeraApp; 