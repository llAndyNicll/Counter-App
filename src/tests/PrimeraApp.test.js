import React from 'react';
import { shallow } from 'enzyme';

import PrimeraApp from '../PrimeraApp';

describe( 'Pruebas del archivo PrimeraApp.js' , () => {

//    test( 'Debe de mostrar el mensaje "Hola, Soy Goku".' , () => {

//        const saludo = 'Hola, Soy Goku';

//        const { getByText } = render( <PrimeraApp saludo = { saludo } /> );

//        expect( getByText( saludo ) ).toBeInTheDocument();

//    });

    test( 'Debe de mostrar <PrimeraApp /> corerrectamente.' , () => {

        const saludo = 'Hola, Soy Goku';

        const wrapper = shallow( <PrimeraApp saludo = { saludo } /> );

        expect( wrapper ).toMatchSnapshot();

    });

    test( 'Debe de mostrar el subtitulo enviado por props.' , () => {

        const saludo = 'Hola, Soy Goku';

        const subtitulo = 'Soy un subtitulo';

        const wrapper = shallow( 
            <PrimeraApp 
                saludo = { saludo } 
                subtitulo = { subtitulo }
            /> );

        const textoParrafo = wrapper.find('p').text();

        // console.log( textoParrafo );

        expect( textoParrafo ).toBe( subtitulo );

    });

});