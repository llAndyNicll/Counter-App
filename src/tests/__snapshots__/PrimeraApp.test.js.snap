// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del archivo PrimeraApp.js Debe de mostrar <PrimeraApp /> corerrectamente. 1`] = `
<div>
  <h1>
     
    Hola, Soy Goku
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</div>
`;
